includes/class-branding.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * The Class for adding branding to the dashboard.
+ *
+ * @package aspire-update
+ */
+
+namespace AspireUpdate;
+
+/**
+ * The Class for adding branding to the dashboard.
+ */
+class Branding {
+	/**
+	 * Hold a single instance of the class.
+	 *
+	 * @var object
+	 */
+	private static $instance = null;
+
+	/**
+	 * The Constructor.
+	 */
+	public function __construct() {
+		$admin_settings = Admin_Settings::get_instance();
+		if ( $admin_settings->get_setting( 'enable', false ) ) {
+			add_action( 'admin_notices', [ $this, 'output_admin_notice' ] );
+		}
+	}
+
+	/**
+	 * Initialize Class.
+	 *
+	 * @return object
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Output admin notice.
+	 *
+	 * @return void
+	 */
+	public function output_admin_notice() {
+		if ( defined( 'AP_REMOVE_BRANDING' ) && AP_REMOVE_BRANDING ) {
+			return;
+		}
+
+		$current_screen = get_current_screen();
+		if ( ! $current_screen instanceof \WP_Screen ) {
+			return;
+		}
+
+		$message = '';
+		switch ( $current_screen->base ) {
+			case 'plugins':
+			case 'plugin-install':
+				$message = sprintf(
+					/* translators: 1: The name of the plugin, 2: The documentation URL. */
+					__( 'Your plugin updates are now powered by <strong>%1$s</strong>. <a href="%2$s">Learn more</a>', 'AspireUpdate' ),
+					'AspireUpdate',
+					__( 'https://docs.aspirepress.org/aspireupdate/', 'AspireUpdate' )
+				);
+				break;
+			case 'themes':
+			case 'theme-install':
+				$message = sprintf(
+					/* translators: 1: The name of the plugin, 2: The documentation URL. */
+					__( 'Your theme updates are now powered by <strong>%1$s</strong>. <a href="%2$s">Learn more</a>', 'AspireUpdate' ),
+					'AspireUpdate',
+					__( 'https://docs.aspirepress.org/aspireupdate/', 'AspireUpdate' )
+				);
+				break;
+			case 'update-core':
+				$message = sprintf(
+					/* translators: 1: The name of the plugin, 2: The documentation URL. */
+					__( 'Your WordPress, plugin, theme and translation updates are now powered by <strong>%1$s</strong>. <a href="%2$s">Learn more</a>', 'AspireUpdate' ),
+					'AspireUpdate',
+					__( 'https://docs.aspirepress.org/aspireupdate/', 'AspireUpdate' )
+				);
+				break;
+		}
+
+		if ( '' === $message ) {
+			return;
+		}
+
+		$icon         = file_get_contents( plugin_dir_path( __DIR__ ) . 'assets/images/aspirepress-logo-icon.svg' );
```

Comment:
Can we set this up in CSS?  file_get_contents is not always supported on hosts and can also create unnecessary permission errors.